src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -277,12 +335,28 @@ bool CartGridStrategy::handleGcode(Gcode *gcode)
             // first wait for an empty queue i.e. no moves left
             THEKERNEL->conveyor->wait_for_idle();
 
-            int n = gcode->has_letter('I') ? gcode->get_value('I') : 0;
-            float x = x_size, y = y_size;
-            if(gcode->has_letter('X')) x = gcode->get_value('X'); // override default probe width
-            if(gcode->has_letter('Y')) y = gcode->get_value('Y'); // override default probe length
-            if(n == 0) n = 7;
-            probe_grid(n, x, y, gcode->stream);
+            int n = gcode->has_letter('I') ? gcode->get_value('I') : configured_grid_x_size;
+            int m = gcode->has_letter('J') ? gcode->get_value('J') : configured_grid_y_size;
```

Comment:
So J is now compulsory for this gcode? This doesn't follow very easily from the way that grid size is overridden. Example:
Probing a 5x5 grid, saved config grid is 7x7. 
We're now required to specify I5 J5, instead of just I5 like before. Specifying I5 only would get a grid of 5x7, which is probably not is expected. 

There should be an error check here at least if grid_size is defined or completely to avoid this pitfall if both I and J are compulsory for probing to work.